backend/coins/btc/maketx/txsize_internal_test.go

```diff
@@ -87,20 +90,17 @@ func TestEstimateTxSize(t *testing.T) {
 				}
 
 				estimatedSize := estimateTxSize(
-					len(tx.TxIn),
-					inputAddress.Configuration,
+					inputConfigurations,
 					len(outputPkScript), changePkScriptSize)
 				require.Equal(t, mempool.GetTxVirtualSize(btcutil.NewTx(tx)), int64(estimatedSize))
 			})
 	}
 
-	for _, inputScriptType := range scriptTypes {
-		for _, outputScriptType := range scriptTypes {
-			test(inputScriptType, outputScriptType, nil)
-			for _, changeScriptType := range scriptTypes {
-				changeScriptType := changeScriptType // avoids referencing the same variable across loop iterations
-				test(inputScriptType, outputScriptType, &changeScriptType)
-			}
+	for _, outputScriptType := range scriptTypes {
+		test(outputScriptType, nil)
+		for _, changeScriptType := range scriptTypes {
+			changeScriptType := changeScriptType // avoids referencing the same variable across loop iterations
```

Comment:
i dislike this very much, I think we had the same discussion before maybe? It is only because Go's type system sucks that you can assign it `""`. If it was possible, it would be an enum.

I like to treat types like this like enums and not assign them values that are not explicitly listed as valid.

Reply:
Right, and `changeScriptType := changeScriptType` doesn't suck, esp. if it can be avoided here?

Maybe we should switch to Rust or Haskell then, because bending a language to fit something else only makes it worse imho.

Reply:
Not sure what to say, in Python I also try to stick to one type per variable, even though I can change the type all the time. Just because I can does not mean I should?

> Right, and changeScriptType := changeScriptType doesn't suck, esp. if it can be avoided here?

True, that sucks massively. Even convinced me to make an exception ;) fixed

Reply:
@x1ddos with going from `test()` to `t.Run()` with a closure, I need to add back the shadow var hack again :(

https://github.com/digitalbitbox/bitbox-wallet-app/pull/858/commits/910601f53cc2958633819d694a555d448bee4a21

Really one of the strangest and most annoying Go quirks.

Reply:
Hahaha :laughing: 

I totally get the point about types. To make a compromise, I would've actually had another arg to the test func:

```go
// testEstimate runs the actual TX size estimate test.
// If the addChange is false, the changeScriptType arg is ignored and the TX under test has no change output.
testEstimate(t *testing.T, useSegwit, addChange bool, outputScriptType, changeScriptType signing.ScriptType) { 
  ...
}
```

then there's no need for an empty `changeScriptType` string arg.